### ISSUE TITLE
[WMS check] reporting

### ISSUE THREAD

**Issue by MaelREBOUX** [georchestra/sdi-consistence-check#30]
Done from terminal in Ubuntu

```
python checker.py --mode WMS --server https://public.sig.rennesmetropole.fr/geoserver/wms/


No ~/.sdichecker file found, skipping credentials definition.

SDI check


mode: WMS

WxS service URL: https://public.sig.rennesmetropole.fr/geoserver/wms/
output mode: log

start time: 2017-03-27 14:25:53



#0
  Layer: ref_fonds:cim
  Error: No metadata defined for layer ref_fonds:cim

#1
  Layer: ref_fonds:pvci
  Error: No metadata defined for layer ref_fonds:pvci

#2
  Layer: ref_fonds:pvci_fondplan
  Error: No metadata defined for layer ref_fonds:pvci_fondplan

#3
  Layer: ref_fonds:pvci_nb
  Error: No metadata defined for layer ref_fonds:pvci_nb

#4
  Layer: ref_fonds:pvci_nomcommunes
  Error: No metadata defined for layer ref_fonds:pvci_nomcommunes

#5
  Layer: ref_fonds:pvci_toponymie
  Error: No metadata defined for layer ref_fonds:pvci_toponymie

#6
  Layer: ref_fonds:pvci_v_troncon_test
  Error: No metadata defined for layer ref_fonds:pvci_v_troncon_test

#7
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=128352bb-1573-4b82-8248-9704f5bf0d9f not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#8
  Layer: ref_fonds:pvci_syntplu_za
  Error: No metadata defined for layer ref_fonds:pvci_syntplu_za

#9
  Layer: ref_fonds:pvci_tu_rm
  Error: No metadata defined for layer ref_fonds:pvci_tu_rm

#10
  Layer: ref_fonds:pvci_v_troncon_lgn
  Error: No metadata defined for layer ref_fonds:pvci_v_troncon_lgn

#11
  Layer: ref_fonds:rte_bretagne_ligne
  Error: No metadata defined for layer ref_fonds:rte_bretagne_ligne

#12
  Layer: ref_fonds:v_axe_rail
  Error: No metadata defined for layer ref_fonds:v_axe_rail

#13
  Layer: ref_fonds:v_voie_lgn
  Error: No metadata defined for layer ref_fonds:v_voie_lgn

#14
  Layer: raster:ortho2014
  Error: No metadata defined for layer raster:ortho2014

#15
  Layer: None
  Error: Metadata https://sig.agglo-rennesmetropole.fr/geonetwork/srv/fre/catalog.search#/metadata/3e547210-2e63-4295-91f5-fb89df6cc99f not found or invalid for layer 'None': Unable to parse the text/html metadata: HTTPSConnectionPool(host='sig.agglo-rennesmetropole.fr', port=443): Max retries exceeded with url: /geonetwork/srv/fre/catalog.search (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Unavailable',)))

#16
  Layer: None
  Error: Metadata https://sig.agglo-rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=3e547210-2e63-4295-91f5-fb89df6cc99f not found or invalid for layer 'None': Unable to parse the text/xml metadata: HTTPSConnectionPool(host='sig.agglo-rennesmetropole.fr', port=443): Max retries exceeded with url: /geonetwork/srv/fre/xml.metadata.get?uuid=3e547210-2e63-4295-91f5-fb89df6cc99f (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Unavailable',)))

#17
  Layer: None
  Error: Metadata https://sig.agglo-rennesmetropole.fr/geonetwork/srv/fre/catalog.search#/metadata/f160f512-f163-467d-8841-930dbb6eba9c not found or invalid for layer 'None': Unable to parse the text/html metadata: HTTPSConnectionPool(host='sig.agglo-rennesmetropole.fr', port=443): Max retries exceeded with url: /geonetwork/srv/fre/catalog.search (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Unavailable',)))

#18
  Layer: None
  Error: Metadata https://sig.agglo-rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=f160f512-f163-467d-8841-930dbb6eba9c not found or invalid for layer 'None': Unable to parse the text/xml metadata: HTTPSConnectionPool(host='sig.agglo-rennesmetropole.fr', port=443): Max retries exceeded with url: /geonetwork/srv/fre/xml.metadata.get?uuid=f160f512-f163-467d-8841-930dbb6eba9c (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 503 Service Unavailable',)))

#19
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=31cd10dc-1245-4554-8ca9-232f925d4402 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#20
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=35dac6d8-b573-4a5a-ad98-541296863e99 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#21
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=be36bcbf-9d0e-4425-9b8b-c58c155c96e7 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#22
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=uuid=33f019b4-40e5-4fe7-9bc4-358960e021bb not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#23
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=dc03f7c6-d6f3-4b8f-8491-e97ba77defba not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#24
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=53056e53-854c-48c5-8908-0e0492f79c61 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#25
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=b66e4413-94dc-4a7e-854c-77d39c1824c0 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#26
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=e4b2583c-3033-4d28-8196-5b93db3d2482 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#27
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=89695a54-d838-42fd-bd3b-10413d521b85 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#28
  Layer: None
  Error: Metadata https://public.sig.rennesmetropole.fr/geonetwork/srv/fre/catalog.search#/metadata/9e4e8a3d-190d-4397-8d24-97aa37d1b3d7 not found or invalid for layer 'None': Unable to parse the text/xml metadata: mismatched tag: line 19, column 5

#29
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=c2b005bf-5814-4038-aa18-8c9badb4041a not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#30
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=264f362a-8552-4689-af57-10de9a0c1594 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#31
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=8dfdbb9a-29ca-4780-b767-7b2aa3ab5d2a not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#32
  Layer: None
  Error: Metadata https://public.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=03268497-3416-4326-9119-48744df1a3c2 not found or invalid for layer 'None': 'text/xml' Metadata not found (HTTP 500): 500 Server Error: Internal Server Error for url: https://public.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=03268497-3416-4326-9119-48744df1a3c2

#33
  Layer: urba_docs:DOC_URBA
  Error: No metadata defined for layer urba_docs:DOC_URBA

#34
  Layer: urba_docs:DOC_URBA_COM
  Error: No metadata defined for layer urba_docs:DOC_URBA_COM

#35
  Layer: urba_docs:ZONE_URBA
  Error: No metadata defined for layer urba_docs:ZONE_URBA

#36
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=79e2f4b2-fb74-4215-ab57-5d6cb02814c8 not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#37
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=60fe8ca9-da74-4b3d-a073-5fab635317aa not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23

#38
  Layer: None
  Error: Metadata https://portail.sig.rennesmetropole.fr/geonetwork/srv/fre/xml.metadata.get?uuid=df789a3f-9980-4629-95be-50a8b04eac3c not found or invalid for layer 'None': Unable to parse the text/xml metadata: undefined entity: line 91, column 23



62 layers parsed, 39 inconsistencies found (62 %)
end time: 2017-03-27 14:26:05
```

Real life errors -> to be corrected. A doubt on the %age see #29 

**An issue :  how to found the layer in error with ```Layer: None``` ??**

**Comment by MaelREBOUX** [comment]
The ```Layer``` info must be provided by the WMS Layer Name  /WMS_Capabilities/Capability/Layer/Layer/Name. Actually seems to be provided by the CSW metadata. So 'None' appears when no MD found.
Correct ?

Adding the title of the metadata (/gmd:MD_Metadata/gmd:identificationInfo/gmd:MD_DataIdentification/gmd:citation/gmd:CI_Citation/gmd:title/gco:CharacterString) would be appreciable (fast check by the human reader).

In summary :

```
#{number}
  Layer name : {/WMS_Capabilities/Capability/Layer/Layer/Name}
  Layer title : {/WMS_Capabilities/Capability/Layer/Layer/Title}
  Metadata : error : no metadata defined for the layer | error : the metadata was not found : {url} | ok
  MD title : {/gmd:MD_Metadata/gmd:identificationInfo/gmd:MD_DataIdentification/gmd:citation/gmd:CI_Citation/gmd:title/gco:CharacterString}
```

**Comment by pmauduit** [comment]
The behaviour is differrent from the CSW output: only errors are printed out in case of WMS / WFS.

This should require a code revamp for both modes.

**Comment by pmauduit** [comment]
Cf comment #31

**Comment by catmorales** [comment]
New test this morning on portail-test
194 layers dans le getcapabilities
 `/usr/local/bin/python3.4 checker.py --mode WMS --server https://portail-test.sig.rennesmetropole.fr/geoserver/wms --disable-ssl-verification`
  result is:

#36
  Layer: ref_fonds:mnt_courbe_niveau
  Error: No metadata defined for layer ref_fonds:mnt_courbe_niveau

#36
  Layer: ref_fonds:mnt_point_culminant
  Error: No metadata defined for layer ref_fonds:mnt_point_culminant

 120 layers parsed, 27 inconsistencies found (22 %)
end time: 2017-04-06 11:45:46

````
we have the same problems with the number results as seen yesterday (194 layers in wms capabilities)

**Comment by pmauduit** [comment]
> we have the same problems with the number results as seen yesterday (194 layers in wms capabilities)
> #36

portail-test needs a git update
